Oppdatert fargepalett for nav-frontend-core + ds-tokens

KenAJoh opened the pull request:
- Farger kalkuleres i ds-tokens ved hjelp av denne koden https://github.com/hihayk/scale/blob/69b766bba2db046d3e8cb4026ae32a32c897f9ff/src/utils.js#L62
- Siden less sine `mix` og `saturate` funksjoner leverer helt forskjellige farger enn de som blir kalkulert i ds-tokens, er de satt eksplisitt i `nav-frontend-core`.
- `orangeFocus` og alle lighten/darken variantene er fjernet fra core.
- navRod sine lighten/darken varianter er fjernet fra core
- La til purple og limegreen farger og variasjoner i tokens.
- Oppdaterte Bekreftcheckbokspanel til å bruke `navOransjeLighten80` istedenfor `orangeFocus`
- Oppdaterte nettsiden og diverse fargebruk i repoet til de nye gråfargene.

For å kjøre denne oppdateringen merges PR -> cancel workflow -> kjøre manuell lerna version major + lerna publish fra master branch

KenAJoh committed:
oppdatert tokens farger

KenAJoh committed:
oppdatert farger i core

KenAJoh committed:
fikset gammel bruk av oransjeFocus

KenAJoh committed:
fjernet orangeFocus fra website og docs

KenAJoh committed:
oppdatert fargebruk i diverse pakker2

KenAJoh committed:
fjernet ubrukte tokens

andnorda commented on @navikt/ds-tokens/src/index.js:
```diff
@@ -1,24 +1,39 @@
-const mix = (a, b, percentage) =>
-  Color(a)
-    .mix(Color(b), percentage / 100)
+const lighten = (color, percentage) =>
```
Denne kan vel slettes nå?

KenAJoh replied:
Blir nå brukt for hover-farge i hoved/fare-knapp, men tar å inliner den.

andnorda commented on @navikt/ds-tokens/src/index.js:
```diff
@@ -1,24 +1,39 @@
+
 const white = "#ffffff";
-const darkgray = "#3e3832";
+const darkgray = "#262626";
```
Ser at `#262626` er definert to steder i denne fila nå (navds-color-gray-90). Er det en ryddig måte å unngå det på? Kanskje gråfargene kan defineres på en mer logisk måte nå? F.eks. `Color("black").lighten()` eller no.

KenAJoh replied:
Fant ingen god løsning med .lighten(), men tenker dette er en ok løsning 
```shell
90: { value: "{navds.color.darkgray.value}" },
```

andnorda replied:
Jeg hørte med Wei: Han sa gråfargene var justert for hånd for å sørge for god kontrast, så da gir det mening at det ikke er noen clean måte å generere de på.

andnorda replied:
Jeg kom til å tenke på en ting: Trenger vi egentlig en token som heter darkgray? Betyr ikke `gray-90` "darkgray"?

KenAJoh replied:
Ser for meg det kan være greit å ha `gray-90` som en egen verdi siden den har en spesifikk bruk, men hvis vi går gjennom å spesifiserer tokensene bedre er det kanskje ikke grunn til det lengre. Typ color-text, color-shadow

andnorda commented on @navikt/ds-tokens/src/index.js:
```diff
@@ -36,15 +51,15 @@ module.exports = {
-            80: { value: darken(color, 60) },
-            70: { value: darken(color, 40) },
-            60: { value: darken(color, 20) },
+            90: { value: darkColor(color, 3) },
```
Jeg stemmer for å gå over til samme naming convention som material design og tailwind (50, 100, 200, ..., 800, 900): https://material.io/design/color/the-color-system.html#color-theme-creation https://tailwindcss.com/docs/customizing-colors#overview

Mest fordi fargene i tailwind-implementasjonen min av designsystemet da får navn som passer. Tanker?

andnorda replied:
For våre farger vil det sannsynligvis bli uten 50 først, og så legger vi kanskje til det senere.

KenAJoh replied:
Tenker det er en god ide, ser de fleste systemer bruker lignende navn for sine farger. Hva tenker du om å gjøre denne endringen i en separat PR der vi går gjennom design-tokene vi har til nå å rydder opp i dem. Unngår da at denne PRen har velding mange breaking changes. 
Tenker da å spesifisere dem mer noe lignende https://spectrum.adobe.com/page/design-tokens/ med global/alias/component tokens.

andnorda replied:
Høres ut som en god plan. 👍

andnorda reviewed (commented):
Bra jobba!

KenAJoh committed:
oppdatert gray-90

andnorda committed:
grays in numbers (#1093)

Co-authored-by: Ken <26967723+KenAJoh@users.noreply.github.com>

KenAJoh committed:
Merge branch 'master' into farger-oppdatert

KenAJoh committed:
oppdatert changelog

KenAJoh committed:
oppdatert changelog2

KenAJoh committed:
oppdatert ds-react version

KenAJoh committed:
Merge branch 'master' into farger-oppdatert

KenAJoh committed:
temp oppdatert lerna config

KenAJoh committed:
Lerna versionbump [ci skip]

 - nav-frontend-alertstriper-style@3.0.0
 - nav-frontend-alertstriper@4.0.0
 - nav-frontend-core@6.0.0
 - nav-frontend-ekspanderbartpanel-style@2.0.0
 - nav-frontend-ekspanderbartpanel@4.0.0
 - nav-frontend-etiketter-style@2.0.0
 - nav-frontend-etiketter@3.0.0
 - nav-frontend-grid@2.0.0
 - nav-frontend-hjelpetekst-style@3.0.0
 - nav-frontend-hjelpetekst@3.0.0
 - nav-frontend-ikoner-assets@3.0.0
 - nav-frontend-ikonknapper@2.0.0
 - nav-frontend-knapper-style@2.0.0
 - nav-frontend-knapper@3.0.0
 - nav-frontend-lenkepanel-style@2.0.0
 - nav-frontend-lenkepanel@2.0.0
 - nav-frontend-lenker-style@2.0.0
 - nav-frontend-lenker@2.0.0
 - nav-frontend-lesmerpanel-style@2.0.0
 - nav-frontend-lesmerpanel@2.0.0
 - nav-frontend-lukknapp-style@2.0.0
 - nav-frontend-lukknapp@2.0.0
 - nav-frontend-modal-style@2.0.0
 - nav-frontend-modal@2.0.0
 - nav-frontend-paneler-style@2.0.0
 - nav-frontend-paneler@3.0.0
 - nav-frontend-popover-style@2.0.0
 - nav-frontend-popover@2.0.0
 - nav-frontend-skjema-style@3.0.0
 - nav-frontend-skjema@4.0.0
 - nav-frontend-snakkeboble-style@2.0.0
 - nav-frontend-snakkeboble@3.0.0
 - nav-frontend-spinner@3.0.0
 - nav-frontend-stegindikator-style@2.0.0
 - nav-frontend-stegindikator@2.0.0
 - nav-frontend-tabell-style@2.0.0
 - nav-frontend-tabs-style@2.0.0
 - nav-frontend-tabs@2.0.0
 - nav-frontend-tekstomrade@3.0.0
 - nav-frontend-toggle-style@2.0.0
 - nav-frontend-toggle@2.0.0
 - nav-frontend-typografi-style@2.0.0
 - nav-frontend-typografi@4.0.0
 - nav-frontend-veileder-style@3.0.0
 - nav-frontend-veileder@3.0.0
 - nav-frontend-veilederpanel-style@2.0.0
 - nav-frontend-veilederpanel@2.0.0

KenAJoh committed:
oppdatert lerna config

KenAJoh committed:
oppdatert react version

KenAJoh committed:
oppdatert versions manuelt

KenAJoh committed:
Lerna versionbump [ci skip]

 - @navikt/ds-css@0.4.0
 - @navikt/ds-react@0.4.0
 - @navikt/ds-tokens@0.2.0

KenAJoh committed:
manuelt oppdatert peerDeps

KenAJoh committed:
fikset versioner

KenAJoh committed:
tmp fiks ikoner-assets

KenAJoh committed:
fjernet private on ikoner-assets
